gnocchi/storage/__init__.py

```diff
@@ -329,118 +331,117 @@ def _store_timeserie_splits(self, metric, keys_and_splits,
         # stored in the case that we need to rewrite them fully.
         # First, fetch all those existing splits.
         existing_data = self._get_splits_and_unserialize(
-            metric, [(key, aggregation) for key in keys_to_rewrite])
+            metric, [(key, split.aggregation)
```

Comment:
if the split has the aggregation already, why we passing in the aggregation?

Reply:
The answer is line 367 ;)

Reply:
oh i mean that aggregation is never used anywhere in this method. it only ever uses split.aggregation.

Reply:
Again, aggregation is only passed because it's used as the key of the dict as explained in the comment L367.

Reply:
oh. it's from the other method.